### ISSUE TITLE
Problems with Rshiny deploy and Rgee

### ISSUE THREAD

**Issue by renanmacedosem** [r-spatial/rgee#315]
* rgee version: 1.1.5
* R version: 4.2.2
* Operating System: Ubuntu 20.04

#### At submit an issue, please attached the following information of your `rgee` session:

- [ ] You have the Python API installed (from terminal):
```
earthegine-api: 0.1.340
```

- [ ] You can find the credentials file on your system: 
```r
> library(rgee)
> ee_path <- path.expand("~/.config/earthengine/credentials")
> file.exists(ee_path)
[1] TRUE
```

- [ ] You can run a simple EE command from R: 
```

library(rgee)

# Initialize the Earth Engine module.
ee_Initialize()
```
```
# Print metadata for a DEM dataset.
print(ee$Image('USGS/SRTMGL1_003')$getInfo())


```
output:
```

$type
[1] "Image"

$bands
$bands[[1]]
$bands[[1]]$id
[1] "elevation"

$bands[[1]]$data_type
$bands[[1]]$data_type$type
[1] "PixelType"

$bands[[1]]$data_type$precision
[1] "int"

$bands[[1]]$data_type$min
[1] -32768

$bands[[1]]$data_type$max
[1] 32767


$bands[[1]]$dimensions
[1] 1296001  417601

$bands[[1]]$crs
[1] "EPSG:4326"

$bands[[1]]$crs_transform
$bands[[1]]$crs_transform[[1]]
[1] 0.0002777778

$bands[[1]]$crs_transform[[2]]
[1] 0

$bands[[1]]$crs_transform[[3]]
[1] -180.0001

$bands[[1]]$crs_transform[[4]]
[1] 0

$bands[[1]]$crs_transform[[5]]
[1] -0.0002777778

$bands[[1]]$crs_transform[[6]]
[1] 60.00014




$id
[1] "USGS/SRTMGL1_003"

$version
[1] -1000009435

$properties
$properties$`system:visualization_0_min`
[1] "0.0"

$properties$type_name
[1] "Image"

$properties$keywords
[1] "dem"         "elevation"   "geophysical" "nasa"        "srtm"        "topography"  "usgs"       

$properties$thumb
[1] "https://mw1.google.com/ges/dd/images/SRTM90_V4_thumb.png"

$properties$description
[1] "<p>The Shuttle Radar Topography Mission (SRTM, see <a href=\"https://onlinelibrary.wiley.com/doi/10.1029/2005RG000183/full\">Farr\net al. 2007</a>)\ndigital elevation data is an international research effort that\nobtained digital elevation models on a near-global scale. This\nSRTM V3 product (SRTM Plus) is provided by NASA JPL\nat a resolution of 1 arc-second (approximately 30m).</p><p>This dataset has undergone a void-filling process using open-source data\n(ASTER GDEM2, GMTED2010, and NED), as opposed to other versions that\ncontain voids or have been void-filled with commercial sources.\nFor more information on the different versions see the\n<a href=\"https://lpdaac.usgs.gov/documents/13/SRTM_Quick_Guide.pdf\">SRTM Quick Guide</a>.</p><p>Documentation:</p><ul><li><p><a href=\"https://lpdaac.usgs.gov/documents/179/SRTM_User_Guide_V3.pdf\">User&#39;s Guide</a></p></li><li><p><a href=\"https://lpdaac.usgs.gov/documents/13/SRTM_Quick_Guide.pdf\">General Documentation</a></p></li><li><p><a href=\"https://doi.org/10.1029/2005RG000183\">Algorithm Theoretical Basis Document (ATBD)</a></p></li></ul><p><b>Provider: <a href=\"https://cmr.earthdata.nasa.gov/search/concepts/C1000000240-LPDAAC_ECS.html\">NASA / USGS / JPL-Caltech</a></b><br><p><b>Bands</b><table class=\"eecat\"><tr><th scope=\"col\">Name</th><th scope=\"col\">Description</th></tr><tr><td>elevation</td><td><p>Elevation</p></td></tr></table><p><b>Terms of Use</b><br><p>Unless otherwise noted, images and video on JPL public\nweb sites (public sites ending with a jpl.nasa.gov address) may\nbe used for any purpose without prior permission. For more information\nand exceptions visit the <a href=\"https://www.jpl.nasa.gov/imagepolicy/\">JPL Image Use Policy site</a>.</p><p><b>Suggested citation(s)</b><ul><li><p>Farr, T.G., Rosen, P.A., Caro, E., Crippen, R., Duren, R., Hensley,\nS., Kobrick, M., Paller, M., Rodriguez, E., Roth, L., Seal, D.,\nShaffer, S., Shimada, J., Umland, J., Werner, M., Oskin, M., Burbank,\nD., and Alsdorf, D.E., 2007, The shuttle radar topography mission:\nReviews of Geophysics, v. 45, no. 2, RG2004, at\n<a href=\"https://doi.org/10.1029/2005RG000183\">https://doi.org/10.1029/2005RG000183</a>.</p></li></ul><style>\n  table.eecat {\n  border: 1px solid black;\n  border-collapse: collapse;\n  font-size: 13px;\n  }\n  table.eecat td, tr, th {\n  text-align: left; vertical-align: top;\n  border: 1px solid gray; padding: 3px;\n  }\n  td.nobreak { white-space: nowrap; }\n</style>"

$properties$source_tags
[1] "nasa" "usgs"

$properties$visualization_0_max
[1] "6000.0"

$properties$title
[1] "NASA SRTM Digital Elevation 30m"

$properties$product_tags
[1] "srtm"        "elevation"   "topography"  "dem"         "geophysical"

$properties$provider
[1] "NASA / USGS / JPL-Caltech"

$properties$visualization_0_min
[1] "0.0"

$properties$visualization_0_name
[1] "Elevation"

$properties$date_range
[1] 1039427584 1989827584

$properties$`system:visualization_0_gamma`
[1] "1.6"

$properties$period
[1] 0

$properties$`system:visualization_0_bands`
[1] "elevation"

$properties$provider_url
[1] "https://cmr.earthdata.nasa.gov/search/concepts/C1000000240-LPDAAC_ECS.html"

$properties$visualization_0_gamma
[1] "1.6"

$properties$sample
[1] "https://mw1.google.com/ges/dd/images/SRTM90_V4_sample.png"

$properties$tags
[1] "dem"         "elevation"   "geophysical" "nasa"        "srtm"        "topography"  "usgs"       

$properties$`system:visualization_0_max`
[1] "6000.0"

$properties$`system:visualization_0_name`
[1] "Elevation"

$properties$`system:asset_size`
[1] -351347924

$properties$visualization_0_bands
[1] "elevation"
```

Attach your Python (reticulate) configuration:

```r
library(reticulate)
py_config()
```

```
python:         /home/renan/.virtualenvs/rgee_py/bin/python
libpython:      /usr/lib/python3.8/config-3.8-x86_64-linux-gnu/libpython3.8.so
pythonhome:     /home/renan/.virtualenvs/rgee_py:/home/renan/.virtualenvs/rgee_py
version:        3.8.10 (default, Nov 14 2022, 12:59:47)  [GCC 9.4.0]
numpy:          /home/renan/.virtualenvs/rgee_py/lib/python3.8/site-packages/numpy
numpy_version:  1.24.2
ee:             /home/renan/.virtualenvs/rgee_py/lib/python3.8/site-packages/ee
NOTE: Python version was forced by RETICULATE_PYTHON
```

### Description

I tried to deploy my Rshiny app into shinyapp.io, but I received this message: "paths should be to files the project directory"  in row 5 and row 20 from the utils.R (I used your template for shiny apps). 

### What I Did
If I run de app.R locally, I do not have issues.
```
set_rgee_dependencies <- function() {
  # 1. Create a Python virtual env
  reticulate::virtualenv_create(
    envname = 'rgee_py',
    python = '/usr/bin/python3'
  )

  # 2. Install rgee and rgeeExtra Python dependencies
  reticulate::virtualenv_install(
    envname = "rgee_py",
    packages = c('numpy', "earthengine-api", "jsbeautifier", "regex")
  )


  # 3. Set a Python Env to use with R
  reticulate::use_virtualenv('rgee_py', required = TRUE)
  system(
    sprintf(
      "echo $'EARTHENGINE_INIT_MESSAGE=\"TRUE\"\nEARTHENGINE_PYTHON=%s' > $HOME/.Renviron",
      path.expand("~/.virtualenvs/rgee_py/bin/python")
    )
  )

  # 4. Copy credentials file to the project folder
  oauth_func_path <- system.file("python/ee_utils.py", package = "rgee")
  utils_py <- rgee:::ee_source_python(oauth_func_path)
  ee_path <- sprintf("%s/ndef", ee_utils_py_to_r(utils_py$ee_path()))
  dir.create(ee_path, showWarnings = FALSE, recursive = TRUE)
  file.copy(from = "credentials", to = ee_path, overwrite = TRUE)


  # 5. Initialize Earth Engine
  ee_Initialize()
}
```
Log (shiny app)

```
2023-02-08T20:28:23.305850+00:00 shinyapps[8251537]: Welcome to the Earth Engine client library for R!
2023-02-08T20:28:23.305897+00:00 shinyapps[8251537]: ----------------------------------------------------------------
2023-02-08T20:28:23.305904+00:00 shinyapps[8251537]: It seems it is your first time using rgee. First off, keep in mind that
2023-02-08T20:28:23.305909+00:00 shinyapps[8251537]: Google Earth Engine is only available to registered users, check the
2023-02-08T20:28:23.305913+00:00 shinyapps[8251537]: official website https://earthengine.google.com/ to get more information.
2023-02-08T20:28:23.305920+00:00 shinyapps[8251537]: Before start coding is necessary to set up a Python environment. Run
2023-02-08T20:28:23.305923+00:00 shinyapps[8251537]: rgee::ee_install() to set up automatically, after that, restart the R
2023-02-08T20:28:23.305934+00:00 shinyapps[8251537]: session to see changes. See more than 300+ examples of rgee at
2023-02-08T20:28:23.305939+00:00 shinyapps[8251537]: https://r-earthengine.github.io/.
2023-02-08T20:28:23.305948+00:00 shinyapps[8251537]: rgee will need to create the environmental variable "EARTHENGINE_INIT_MESSAGE"
2023-02-08T20:28:23.305954+00:00 shinyapps[8251537]: in your system to avoid you recieve this message again.
2023-02-08T20:28:23.305962+00:00 shinyapps[8251537]: Would you like to stop to receive this message? [Y/n]:
2023-02-08T20:28:24.058491+00:00 shinyapps[8251537]: Error in ee_connect_to_py(path = ee_current_version, n = 5) : 
2023-02-08T20:28:24.058529+00:00 shinyapps[8251537]:   The current Python PATH: /usr/bin/python3
2023-02-08T20:28:24.058534+00:00 shinyapps[8251537]: does not have the Python package "earthengine-api" installed. Are you restarted/terminated.
2023-02-08T20:28:24.058538+00:00 shinyapps[8251537]: your R session after install miniconda or run ee_install()?
2023-02-08T20:28:24.058542+00:00 shinyapps[8251537]: If no do it, please try again:
2023-02-08T20:28:24.058546+00:00 shinyapps[8251537]: > ee_install(): To create and set a Python environment with all rgee dependencies.
2023-02-08T20:28:24.058549+00:00 shinyapps[8251537]: > ee_install_set_pyenv(): To set a specific Python environment.
2023-02-08T20:28:24.058563+00:00 shinyapps[8251537]: Calls: local ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
2023-02-08T20:28:24.058566+00:00 shinyapps[8251537]: Execution halted
2023-02-08T20:28:24.058573+00:00 shinyapps[8251537]: Shiny application exiting ...
2023-02-08T20:43:56.597376+00:00 shinyapps[8251537]: Container event from container-7734512: stop
```

Output: 

![Screenshot from 2023-02-08 12-44-06](https://user-images.githubusercontent.com/80788337/217646916-ab5bbbaf-2b35-4a34-ac46-567ca184b3f5.png)
![Screenshot from 2023-02-08 12-44-53](https://user-images.githubusercontent.com/80788337/217647039-f59b7f12-1bfe-4fa2-8689-8aaa233028f8.png)

**Comment by renanmacedosem** [comment]
Hi @ambarja ! 

I found the problem: 

1. Your template is clear: run the deploy.R using the terminal. I was using R connect (from Rstudio) because I had a problem with my token. When I updated my token, I was able to run using the terminal. 
2. Using my terminal, I do not have any problems. 

Your template is working well, the problem was my token that fo the same reason, I needed to change it. 

Thanks!
